model/src/funTest/kotlin/utils/PostgresProvenanceFileStorageFunTest.kt

```diff
@@ -22,6 +22,7 @@ package org.ossreviewtoolkit.model.utils
 import io.kotest.core.spec.style.WordSpec
 import io.kotest.matchers.shouldBe
 
+import java.io.ByteArrayInputStream
```

Comment:
How about adding as "syntactic sugar" a `fun putData(provenance, bytes: ByteArray()` for not making the caller that verbose?

Reply:
You mean just for the test code? Because I think the change it `FileListResolver` is not that verbose.

Reply:
I don't enough about the implementation, maybe it's also a requirement of the JDBC driver or so. But after reading some docs I think it should be possible to remove that again later on. Therefore I have rephrased the commit message so that this is basically an ask of the people working on the ORT server to introduce this to unblock us for now pending later investigation if it can be removed again.

Reply:
hm, there is really no pointer to any API docs which shows that it's needed?

Reply:
Maybe [this](https://www.postgresql.org/docs/7.1/jdbc-lo.html#JDBC-LO-EXAMPLE) is a better link as it refer to Java and not C code. So this

    ps.setBinaryStream(2, fis, file.length());

indeed requires the size of the `FileInputStream` to be passed.

Reply:
No, I was hoping that the new commit message made it clear that this is a requirement of our internal implementation. Unfortunately @oheger-bosch is not available anymore today, he could probably provide more details.

Reply:
Looks like `length` is only needed if you don't want all of the stream to be read.

Reply:
Looking at another implementation of this API, the length is used for double checking that the expected amount of bytes has been read from the stream, e.g. fail if it doesn't match, see https://learn.microsoft.com/en-us/sql/connect/jdbc/reference/setbinarystream-method-int-java-io-inputstream-long?view=sql-server-ver16.

Reply:
I'll leave the decision whether to merge up to you guys, even tough I prefer to wait until there is a clear reason that it is really required. Because all API docs we've seen now seem to provide a way without the length.

Reply:
TBH, I'm also reluctant towards this "dummy" change. @mnonnenmacher, could you maybe explain why it's easier to do this change here, than making the change in the server to not require `length`?

Reply:
To give you a bit background information:
In ORT Server, we have introduced a generic storage API for storing all kinds of data. It can be backed by different implementations. Postgres is one of those, another one could be a file storage like Azure Storage Accounts. When designing the API we had the different possible implementations in mind, and some of them require the size of the data to be known beforehand. This is for instance needed by the [putBlob](https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob?tabs=azure-ad) operation from Azure, where requests require a correct `Content-Length` header.

For Postgres itself, the data size is not required for writing or reading blobs. However, the implementation in ORT Server uses the size information to distinguish between data that can be read into memory and large data that needs to be streamed to the client. If the size is below a configurable threshold, it is read directly and send as response to the client. Otherwise, we need to work with temporary files to buffer the data; this is because Postgres requires that blobs can only be accessed in a transaction; however, when sending the HTTP response to the client, the transaction is already closed.

Reply:
Thanks for the details @oheger-bosch. So if instead of Postgres the Azure backend would be mentioned in the commit message as an example that really *requires* the length (like you do above), and also the information from the last paragraph that ORT might use different "modes" depending on the size of the data would go into the commit message, I'd be ok with this change.

Reply:
I have updated the commit message.